patches/tModLoader/Terraria/ModLoader/ModItem.cs

```diff
@@ -871,12 +871,17 @@ public abstract class ModItem : ModTexturedType
 		}
 
 		/// <summary>
-		/// Allows you to tell the game whether this item is a torch that cannot be placed in liquid, a torch that can be placed in liquid, or a glowstick. This information is used for when the player is holding down the auto-select hotkey.
-		/// </summary>
-		/// <param name="dryTorch">if set to <c>true</c> [dry torch].</param>
-		/// <param name="wetTorch">if set to <c>true</c> [wet torch].</param>
-		/// <param name="glowstick">if set to <c>true</c> [glowstick].</param>
-		public virtual void AutoLightSelect(ref bool dryTorch, ref bool wetTorch, ref bool glowstick) {
+		/// Allows you to determine the color of light and dust that this torch emits when held or dropped on the ground
+		/// </summary>
+		/// <param name="R">The red component of the torch light (0-1)</param>
+		/// <param name="G">The green component of the torch light (0-1)</param>
+		/// <param name="B">The blue component of the torch light (0-1)</param>
+		/// <param name="dustId">The id of the dust that the torch emits</param>
+		public virtual void TorchVFX(out float R, out float G, out float B, out int dustId) {
```

Comment:
Parameters should be lowercase (r, g, b). Also, concider possible use-cases of a GlobalItem hook (and resulting ItemLoader method). This would change calls of the form `ModItem.TorchVFX` to `ItemLoader.TorchVFX`

Reply:
I don't think that this needs a `GlobalItem` hook as it doesn't really add anything that couldn't be done before without `HoldItem` and `PostUpdate` overrides in `ModItem` and those already have `GlobalItem` hooks.